pkg/api/metrics.go

```diff
@@ -28,6 +29,12 @@ func (hs *HTTPServer) handleQueryMetricsError(err error) *response.NormalRespons
 	if errors.Is(err, plugincontext.ErrPluginNotFound) {
 		return response.Error(http.StatusNotFound, "Plugin not found", err)
 	}
+	if errors.Is(err, plugins.ErrPluginNotRegistered) {
+		return response.Err(err)
```

Comment:
I have to say that the difference between `.Err` and  `.Error` is a bit confusing. Took me a while to notice that if I was returning a `errutil.Error` I needed to use the former :sweat_smile:

Reply:
Sorry didn't notice there was comments before I started to review. That's basically why I suggest to use the `ErrOrFallback` to not having to care about this in the API layer.